pkg/server/server_sql.go

```diff
@@ -1373,6 +1374,7 @@ func newSQLServer(ctx context.Context, cfg sqlServerArgs) (*SQLServer, error) {
 		sessionRegistry:                cfg.sessionRegistry,
 		closedSessionCache:             cfg.closedSessionCache,
 		jobRegistry:                    jobRegistry,
+		nodeDialer:                     cfg.podNodeDialer,
```

Comment:
I wonder if we want a better name here since in the context of a MT sql server we have two dialers and `nodeDialer` doesn't make it immediately clear which is which. We could perhaps name this one SQLInstanceDialer.  

I don't mind if we want to punt this to a follow-up.